src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -6,9 +6,31 @@
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH_DATA = 6;
 
     @Override
     public User register(User user) {
-        return null;
+        if (checkLoginIfPresent(user) != null ) {
+            throw new InvalidDataException("User with this login has been already registered");
+        } else if (user.getAge() == null) {
+            throw new InvalidDataException("Age must not be null");
+        } else if (user.getLogin() == null) {
+            throw new InvalidDataException("Login mustn't be null");
+        } else if (user.getPassword() == null) {
+            throw new InvalidDataException("Password mustn't be null");
+        } else if (user.getLogin().length() < MIN_LENGTH_DATA) {
+            throw new InvalidDataException("Login must contain at least 6 characters");
+        } else if (user.getPassword().length() < MIN_LENGTH_DATA) {
+            throw new InvalidDataException("Password must contain at least 6 characters");
+        } else if (user.getAge() < MIN_AGE) {
+            throw new InvalidDataException("The user is not grown-up");
```

Comment:
The exception message 'The user is not grown-up' is not informative enough. It should include the minimum age required for registration, such as 'User age must be at least ' + MIN_AGE + ' years old'.